components/server/ee/src/prebuilds/prebuild-manager.ts

```diff
@@ -66,16 +68,27 @@ export class PrebuildManager {
                 .findActivePrebuiltWorkspacesByBranch(project.id, branch);
             const results: Promise<any>[] = [];
             for (const prebuild of prebuilds) {
-                for (const instance of prebuild.instances) {
-                    log.info(
-                        { userId: user.id, instanceId: instance.id, workspaceId: instance.workspaceId },
-                        "Aborting Prebuild workspace because a newer commit was pushed to the same branch.",
-                    );
-                    results.push(this.workspaceStarter.stopWorkspaceInstance({ span }, instance.id, instance.region));
+                try {
+                    for (const instance of prebuild.instances) {
+                        log.info(
+                            { userId: user.id, instanceId: instance.id, workspaceId: instance.workspaceId },
+                            "Aborting Prebuild workspace because a newer commit was pushed to the same branch.",
+                        );
+                        results.push(
+                            this.workspaceStarter.stopWorkspaceInstance(
+                                { span },
+                                instance.id,
+                                instance.region,
+                                StopWorkspacePolicy.ABORT,
```

Comment:
The main change is, to use this new policy which makes ws-manager not care about backups.